src/pipx/commands/reinstall.py

```diff
@@ -84,7 +84,10 @@ def reinstall_all(
     skip: Sequence[str],
 ) -> ExitCode:
     """Returns pipx exit code."""
-    pipx.shared_libs.shared_libs.upgrade(verbose=verbose)
+    if pipx.shared_libs.shared_libs.is_valid:
```

Comment:
What do you think about moving this check to the intervals of the upgrade method? That was any other calls to upgrade wouldn’t be subject to this issue either.

Reply:
Yeah I thought about that.  It's probably good to do.

I hesitated to do that because it makes the code inside of `shared_libs.py` look almost like a circular reference (even though it's not) because `create` falls back to `upgrade`, and then with that code `upgrade` would fall back to `create`.  It ultimate works out ok in fact.

Reply:
OK, made that fix. 👍